### PR TITLE
[Windows] Fix Entry placeholder position using VerticalTextAlignment

### LINKED ISSUE #7844: Placeholder text alignment not respected on Windows ref #3808
### Description

The exact same behaviour as in #3808
image

image

### Steps to Reproduce
```
<Grid ColumnDefinitions="*, auto">
    <Entry Placeholder="Zadejte rodné číslo..." VerticalTextAlignment="Center"/>
    <Button Grid.Column="1" Text="Hledat"/>
</Grid>
```
I removed some extra code that I think is irrelevant.

### Version with bug

6.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

10.0.22000.0

### Did you find any workaround?

It probably worked at some point since the issue was closed? But I do not know which version...

### Relevant log output

_No response_

### LINKED ISSUE #8770: Placeholder text is not vertically centered on Entry on windows
### Description

If you specify Placeholder text for an Entry on windows it is not vertically centered in the field.

image

ios
image

### Steps to Reproduce

set the Placeholder property.

### Version with bug

6.0.400

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

10.0.17763.0

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix Entry placeholder position using VerticalTextAlignment. This PR include changes to get the placeholder TextBlock and update the VerticalTextAlignment.

image

### Issues Fixed

Fixes #7844 
Fixes #8770

### CODE REVIEW

**Review by mattleibow — COMMENTED** [review]
Is it possible to add a unit test for this?

**Comment by JPfahl** [comment]
Is this ever going to make it into production?  TimePickers and Labels do not align vertically either.

File: src/Core/src/Handlers/Entry/EntryHandler.Windows.cs [hunk 1]
```diff
@@ -64,8 +66,11 @@ protected override void DisconnectHandler(TextBox platformView)
 		public static void MapHorizontalTextAlignment(IEntryHandler handler, IEntry entry) =>
 			handler.PlatformView?.UpdateHorizontalTextAlignment(entry);
 
-		public static void MapVerticalTextAlignment(IEntryHandler handler, IEntry entry) =>
-			handler.PlatformView?.UpdateVerticalTextAlignment(entry);
+		public static void MapVerticalTextAlignment(IEntryHandler handler, IEntry entry)
+		{
+			if (handler is EntryHandler entryHandler)
+				handler.PlatformView?.UpdateVerticalTextAlignment(entry, entryHandler._placeholder);
```
Comment (mattleibow) [inline src/Core/src/Handlers/Entry/EntryHandler.Windows.cs:72]: Is it possible to find the placeholder via the name or view apis? Then we don't need to cast and avoid concrete types in these mappers.

**Review by mattleibow — CHANGES_REQUESTED** [review]
Superseded by https://github.com/dotnet/maui/pull/12059

**Comment by mattleibow** [comment]
Superseded by https://github.com/dotnet/maui/pull/12059

**Pull request closed without merging**